src/main/java/ch/njol/skript/expressions/ExprTarget.java

```diff
@@ -61,80 +67,93 @@ public class ExprTarget extends PropertyExpression<LivingEntity, Entity> {
 	}
 
 	@Nullable
-	EntityData<?> type;
+	private EntityData<?> type;
 
-	@SuppressWarnings("unchecked")
 	@Override
+	@SuppressWarnings("unchecked")
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
 		type = exprs[matchedPattern] == null ? null : (EntityData<?>) exprs[matchedPattern].getSingle(null);
 		setExpr((Expression<? extends LivingEntity>) exprs[1 - matchedPattern]);
 		return true;
 	}
 
 	@Override
-	protected Entity[] get(Event e, LivingEntity[] source) {
-		return get(source, en -> {
-			if (getTime() >= 0 && e instanceof EntityTargetEvent && en.equals(((EntityTargetEvent) e).getEntity()) && !Delay.isDelayed(e)) {
-				Entity target = ((EntityTargetEvent) e).getTarget();
+	protected Entity[] get(Event event, LivingEntity[] source) {
+		return get(source, entity -> {
+			if (event instanceof EntityTargetEvent && entity.equals(((EntityTargetEvent) event).getEntity()) && !Delay.isDelayed(event)) {
+				Entity target = ((EntityTargetEvent) event).getTarget();
 				if (target == null || type != null && !type.isInstance(target))
 					return null;
 				return target;
 			}
-			return getTarget(en, type);
+			return getTarget(entity, type);
 		});
 	}
 
 	@Override
-	public Class<? extends Entity> getReturnType() {
-		return type != null ? type.getType() : Entity.class;
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
```

Comment:
Missing all switch cases. Should include a default which returns super.acceptChange. Optionally adding all cases, but a default is needed. This displays a warning.